crates/fluvio-test/src/tests/smoke/mod.rs

```diff
@@ -68,14 +77,90 @@ impl TestOption for SmokeTestOption {
 //}
 
 #[fluvio_test(topic = "test")]
-pub async fn smoke(mut test_driver: Arc<FluvioTestDriver>, mut test_case: TestCase) -> TestResult {
+pub fn smoke(mut test_driver: FluvioTestDriver, mut test_case: TestCase) -> TestResult {
     println!("Starting smoke test");
     let smoke_test_case = test_case.into();
 
-    let start_offsets = produce::produce_message(test_driver.clone(), &smoke_test_case).await;
-    // println!("start sleeping");
-    // fluvio_future::timer::sleep(Duration::from_secs(40)).await;
-    // sleep(Duration::from_secs(40));
-    // println!("end sleeping");
-    consume::validate_consume_message(test_driver.clone(), &smoke_test_case, start_offsets).await;
+    // We're going to handle the `--consumer-wait` flag in this process
+    let producer_process = match fork() {
+        Ok(Fork::Parent(child_pid)) => child_pid,
```

Comment:
this is not show stopper, but this could use regular macro to automate fork like this:
```
#[macro_export]
macro_rules! test_fork! {
    ($child:expr) => {{
         match fork() {
              Ok(Fork::Parent(child_pid)) => child_pid,
              Ok(Fork::Child) => {
                run_block_on($child)
              }
...
        }
    }};
}

```

So this can be written as:
```
  test_fork!(async {
            println!("Producer about to connect");
           let mut test_driver_consumer_wait = test_driver.clone();
          ...

 }
```

Reply:
I can play around with this pattern. The only part I don't want to be overconfident about is if it would be simple enough to join on the pid. Seems like it should be straightforward, and would make this pattern much easier to use.